app/frontend/actions.ts

```diff
@@ -175,11 +163,15 @@ export default ({setState, getState}: {setState: SetStateFn; getState: GetStateF
       const autoLogin = state.autoLogin
       const ticker2Id = null
       const shouldShowPremiumBanner =
-        state.shouldShowPremiumBanner && walletInfo.balance > PREMIUM_MEMBER_BALANCE_TRESHHOLD
-      const isBigDelegator = walletInfo.balance > BIG_DELEGATOR_THRESHOLD
+        state.shouldShowPremiumBanner && PREMIUM_MEMBER_BALANCE_TRESHHOLD < totalWalletBalance
+      const isBigDelegator = totalWalletBalance > BIG_DELEGATOR_THRESHOLD
       setState({
+        accounts: accountsInfo,
+        totalWalletBalance,
+        totalRewardsBalance,
+        shouldShowSaturatedBanner,
         walletIsLoaded: true,
-        ...walletInfo,
+        ...accountsInfo[0],
```

Comment:
I'd rather pass the full accountsInfo and have the current account index set in one specific property of the state to prevent the scenario that some parts of the state are associated with one account and some with another

Reply:
I agree, previously I implemented the multi-account support as quickly as possible and this is definitely one of the early designs which enabled me to implement this quickly without a lot of changes. I wanted to left some state-related refactoring to another PR and this would be probably a part of it but the fact that we changed the design and using the transfer modal a user can switch "source" account rather quickly, it makes a lot more sense to do it right now.

An alternative would be to make sure we don't use the "general" state for tx calculation etc and only use it for displaying (in which case we don't care that much about how quickly it changes). When using something from the state for txs (like balance) we should always refer to it like `state.accounts[account.accountIndex].balance)` . (`account` is a global variable that changes synchronously so its even better). 

Also implementing both of these approaches would be nice. We can discuss this tomorrow and I ll implement at least one of them :+1:

Reply:
I'm not sure if I get your second approach - if we use the "general" state when displaying, wouldn't a "general" state out of sync potentially mislead the user into thinking the balance of their account is different than it actually is? And wouldn't it be confusing if the funds got send from a different account than the one they see?

Reply:
I still think we should do the approach you suggested and even better would be to do both. I ll implement your approach tomorrow. But to answer your question, the more important thing is to use the correct state when creating txs thus never use the "general" state in those cases. But in the end, the general state is also almost up to date. 

TLDR: we should use `state.account[account.accountIndex]` which changes synchronously when using state for building tx a and we don't really need to worry about using "general" state everywhere else since it's only milliseconds late. This would be a middle way between (for example) refactoring everything from `state.txHistory` to `state. accounts[state.selectedAccount].txHistory`and doing nothing. :D

Reply:
I'm not really worried about a delay between the global state and the local account state, but rather a potential bug that may cause "artifacts" from a previously selected account to persist in the state and be shown to the user for a different account. This would be particularly nasty as it would manifest only when switching accounts so users may report very weird problems until we figure out what exactly is going on

Reply:
I agree and I m not aware of any blocker so I'll implement your approach I just wanted to point out the second one and why it beneficial as well.